lms/djangoapps/ora_staff_grader/views.py

```diff
@@ -151,29 +191,35 @@ class InitializeView(StaffGraderBaseView):
     Errors:
     - MissingParamResponse (HTTP 400) for missing params
     - BadOraLocationResponse (HTTP 400) for bad ORA location
+    - UnknownError (HTTP 500) for ORA/platform internal errors
     """
     @require_params([PARAM_ORA_LOCATION])
     def get(self, request, ora_location, *args, **kwargs):
-        response_data = {}
-
-        # Get ORA block
         try:
+            response_data = {}
+
+            # Get ORA block
             ora_usage_key = UsageKey.from_string(ora_location)
             response_data['oraMetadata'] = modulestore().get_item(ora_usage_key)
-        except (InvalidKeyError, ItemNotFoundError):
-            return BadOraLocationResponse()
 
-        # Get course metadata
-        course_id = str(ora_usage_key.course_key)
-        response_data['courseMetadata'] = get_course_overview_or_none(course_id)
+            # Get course metadata
+            course_id = str(ora_usage_key.course_key)
+            response_data['courseMetadata'] = get_course_overview_or_none(course_id)
 
-        # Get list of submissions for this ORA
-        response_data['submissions'] = self.get_submissions(request, ora_location)
+            # Get list of submissions for this ORA
+            response_data['submissions'] = self.get_submissions(request, ora_location)
 
-        # Get the rubric config for this ORA
-        response_data['rubricConfig'] = self.get_rubric_config(request, ora_location)
+            # Get the rubric config for this ORA
+            response_data['rubricConfig'] = self.get_rubric_config(request, ora_location)
 
-        return Response(InitializeSerializer(response_data).data)
+            return Response(InitializeSerializer(response_data).data)
+
+        # Catch bad ORA location
+        except (InvalidKeyError, ItemNotFoundError):
+            return BadOraLocationResponse()
```

Comment:
Shouldn't this be on the other views that accept a location as well?

Reply:
Not necessarily, there are 2 slightly different error paths:

1. This is the only endpoint that actually reads from the ORA item from the platform side. If it fails to find the ORA we need to know before we try to poll any of its data.
2. The other endpoints use XBlock handlers exclusively which, if passed a bad ORA location, return a 404. We could catch just those, but they would be through the XBlock handler wrapping code instead (to save us a modulestore lookup)